docs/topics/multiplatform/multiplatform-compatibility-guide.md

```diff
@@ -192,4 +192,47 @@ The available scopes are `Api`, `Implementation`, `CompileOnly`, and `RuntimeOnl
 
 In Kotlin 1.8.0, an error is introduced when using old configuration names in hard-coded strings.
 
-For more information, see the [corresponding issue in YouTrack](https://youtrack.jetbrains.com/issue/KT-35916/).
\ No newline at end of file
+For more information, see the [corresponding issue in YouTrack](https://youtrack.jetbrains.com/issue/KT-35916/).
+
+## Gradle properties for hierarchical support
+
+<anchor name="deprecate-hmpp-properties"></anchor>
+
+**What's changed?**
+
+Throughout its evolution, Kotlin Multiplatform was gradually introducing the support for hierarchical structure,
+an ability to have intermediate source sets between the common source set `commonMain` and any platform-specific one,
+for example, `jvmMain`.
+
+For the transition period, while the toolchain wasn't stable enough, a couple of Gradle properties were introduced,
+allowing granular opt-ins and opt-outs.
+
+Since Kotlin 1.6.20, the hierarchical structure support has been enabled by default. However, these properties were kept
+for opting out in case of blocking issues. After processing all the feedback, we're now starting to phase out those
+properties completely.
+
+The following properties are now deprecated and will be removed in Kotlin 1.9.0:
+
+* `kotlin.internal.mpp.hierarchicalStructureByDefault`
+* `kotlin.mpp.enableCompatibilityMetadataVariant`
+* `kotlin.mpp.hierarchicalStructureSupport`
+* `kotlin.mpp.enableGranularSourceSetsMetadata`
```

Comment:
Realized that `kotlin.native.enableDependencyPropagation` is missing from the list, please add